benchmarks/CMakeLists.txt

```diff
@@ -29,6 +29,12 @@ else()
   target_compile_options(benchmark_main PRIVATE -w)
 ENDIF()
 
+option(ARBORX_ENABLE_CALIPER "Enable Caliper instrumentation" OFF)
+if(ARBORX_ENABLE_CALIPER)
+  find_package(caliper REQUIRED)
+  add_compile_definitions(ARBORX_ENABLE_CALIPER)
```

Comment:
I chose `add_compile_definitions` instead of `target_link_libraries` for ArborX target so that the caliper's usage scope is restricted to benchmarks.